sys/include/net/gnrc/tcp/config.h

```diff
@@ -0,0 +1,160 @@
+ */
+
+/**
+ * @defgroup    net_gnrc_tcp TCP
+ * @ingroup     net_gnrc
+ * @brief       RIOT's tcp implementation for the gnrc stack
+ *
+ * @{
+ *
+ * @file
+ * @brief       TCP configuration, includes buffersizes, timeout durations
+ *
+ * @author      Simon Brummer <simon.brummer@haw-hamburg.de>
+ * @}
+ */
+
+#ifndef GNRC_TCP_CONFIG_H_
+#define GNRC_TCP_CONFIG_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Status Flags for TCP
+ */
+#define GNRC_TCP_STATUS_PASSIVE (1 << 0)
+#define GNRC_TCP_STATUS_ACCEPTED (1 << 1)
+#define GNRC_TCP_STATUS_ALLOW_ANY_ADDR (1 << 2)
+
+/**
+ * @brief Timeout Duration for user calls. Default 2 minutes
+ */
+#ifndef GNRC_TCP_USER_TIMEOUT_DURATION
+#define GNRC_TCP_USER_TIMEOUT_DURATION (2 * 60 * 1000 * 1000)
+#endif
+
+/**
+ * @brief Maximum Segment Lifetime. Default 30 secounds
+ */
+#ifndef GNRC_TCP_MSL
+#define GNRC_TCP_MSL (30 * 1000 * 1000)
+#endif
+
+/**
+ * @brief   Message queue size for the TCP handling thread
+ */
+#ifndef GNRC_TCP_MSG_QUEUE_SIZE
+#define GNRC_TCP_MSG_QUEUE_SIZE (8U)
+#endif
+
+/**
+ * @brief   Priority of the tcp handling thread, must be lower than the applications prio.
+ */
+#ifndef GNRC_TCP_PRIO
+#define GNRC_TCP_PRIO (THREAD_PRIORITY_MAIN - 2U)
+#endif
+
+/**
+ * @brief   Default stack size for the TCP handling thread
+ */
+#ifndef GNRC_TCP_STACK_SIZE
+#define GNRC_TCP_STACK_SIZE (THREAD_STACKSIZE_DEFAULT)
+#endif
+
+/* Maximum Segment Size */
+/* If IPv6 is used. Get MSS = 1280 - IPv6-Hdr - TCP-Hdr = 1220*/
+#ifdef  MODULE_GNRC_IPV6
+#ifndef GNRC_TCP_MSS
+#define GNRC_TCP_MSS (1220U)
+#endif
+#endif
+
+/* Default MSS. Maximum amount of payload in bytes */
+#ifndef GNRC_TCP_MSS
+#define GNRC_TCP_MSS (576U)
+#endif
+
+/* MSS Multiplicator = Number of MSS sized packets stored in receive buffer */
+#ifndef GNRC_TCP_MSS_MULTIPLICATOR
+#define GNRC_TCP_MSS_MULTIPLICATOR (1U)
+#endif
+
+/* Default Window Size */
+#ifndef GNRC_TCP_DEFAULT_WINDOW
+#define GNRC_TCP_DEFAULT_WINDOW (GNRC_TCP_MSS * GNRC_TCP_MSS_MULTIPLICATOR)
+#endif
+
+/* Number of Preallocated receive buffers */
+#ifndef GNRC_TCP_RCV_BUFFERS
+#define GNRC_TCP_RCV_BUFFERS 1
+#endif
+
+/* Default Receive Buffer Size */
+#ifndef GNRC_TCP_RCV_BUF_SIZE
+#define GNRC_TCP_RCV_BUF_SIZE (GNRC_TCP_DEFAULT_WINDOW)
+#endif
+
+/* Lower Bound for RTO = 1 sec (see RFC 6298) */
+#ifndef GNRC_TCP_RTO_LOWER_BOUND
+#define GNRC_TCP_RTO_LOWER_BOUND (1000 * 1000)
+#endif
+
+/* Upper Bound for RTO = 60 sec (see RFC 6298) */
+#ifndef GNRC_TCP_RTO_UPPER_BOUND
+#define GNRC_TCP_RTO_UPPER_BOUND (60 * 1000 * 1000)
+#endif
+
+/* Assumes clock granularity for TCP of 10 ms (see RFC 6298) */
+#ifndef GNRC_TCP_RTO_GRANULARITY
+#define GNRC_TCP_RTO_GRANULARITY (10 * 1000)
+#endif
+
+/* Alpha value for RTO calculation, default is 1/8 */
+#ifndef GNRC_TCP_RTO_A
+#define GNRC_TCP_RTO_A (0.125f)
+#endif
+
+/* Beta value for RTO calculation, default is 1/4 */
+#ifndef GNRC_TCP_RTO_B
+#define GNRC_TCP_RTO_B (0.25f)
+#endif
```

Comment:
I would rather not use floating point arithmetic. Scale values accordingly and adapt math ops. That is, instead of `foo = bar * GNRC_TCP_RTO_A` make it `foo = bar / GNRC_TCP_RTO_A` with `#define GNRC_TCP_RTO_A (8U)`.